ql/src/java/org/apache/hadoop/hive/ql/io/parquet/ParquetRecordReaderBase.java

```diff
@@ -209,6 +214,22 @@ public FilterCompat.Filter setFilter(final JobConf conf, MessageType schema) {
     }
   }
 
+  private MessageType getSchemaWithoutPartitionColumns(JobConf conf, MessageType schema) {
+    String partCols = conf.get(IOConstants.PARTITION_COLUMNS);
+    if (partCols != null && partCols.length() > 0) {
```

Comment:
```suggestion
    if (partCols != null && !partCols.isEmpty()) {
```
Nit: maybe using isEmpty here can help?